pygis/docs/e_extraction.md

```diff
@@ -0,0 +1,350 @@
+
+Sources: [Indexing and Selecting Data, GeoPandas](https://geopandas.org/docs/user_guide/indexing.html); [Indexing and selecting data, pandas](https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html); [How do I select a subset of a DataFrame?, pandas](https://pandas.pydata.org/pandas-docs/stable/getting_started/intro_tutorials/03_subset_data.html)
+
+We will use use different criteria to subset the wells dataset.
+
+```{code-cell} ipython3
+# Display attribute table
+display(wells)
+```
+
+The criteria can use a variety of operators, including comparison and logical operators.
+
+```{code-cell} ipython3
+# Select wells that are public supply
+wells_subset_1 = wells[(wells["WELL_USE"] == "Public Supply")]
+
+# Display first five and last five rows of attribute table
+display_table(table_name = "San Francisco Bay Area Wells - Public Supply", attribute_table = wells_subset_1)
+```
+
+```{code-cell} ipython3
+# Select wells that are public supply and have a depth greater than 50 ft
+wells_subset_2 = wells[(wells["WELL_USE"] == "Public Supply") & (wells["WELL_DEPTH"] > 50)]
+
+# Display first five and last five rows of attribute table
+display_table(table_name = "San Francisco Bay Area Wells - Public Supply with Depth Greater than 50 ft", attribute_table = wells_subset_2)
+```
+
+```{code-cell} ipython3
+# Select wells that are public supply and have a depth greater than 50 ft OR are residential
+wells_subset_3 = wells[((wells["WELL_USE"] == "Public Supply") & (wells["WELL_DEPTH"] > 50)) | (wells["WELL_USE"] == "Residential")]
+
+# Display first five and last five rows of attribute table
+display_table(table_name = "San Francisco Bay Area Wells - Public Supply with Depth Greater than 50 ft or Residential", attribute_table = wells_subset_3)
+```
+
+## Select by Location
+
+Selecting by location selects features based on its relative spatial relationship with another dataset.
+
+There are multiple spatial relationships available in `geopandas`.
+
+| Spatial Relationship | Function(s) |
+| :------------ | ---------------------------------------: |
+| contains | `contains()` |
+| covers | `covers()` |
+| covered by | `covered_by()` |
+| crosses | `crosses()` |
+| disjoint | `disjoint()` |
+| equal geometry | `geom_equals()`, `geom_almost_equals()`, `geom_equals_exact()` |
+| intersects | `intersects()` |
+| overlaps | `overlaps()` |
+| touches | `touches()` |
+| within | `within()` |
+
+Source: [GeoSeries - Binary Predicates, GeoPandas](https://geopandas.org/docs/reference/geoseries.html#binary-predicates)
+
+The functions for these spatial relationships will generally output a `pandas` series with Boolean values (`True` or `False`) whose indexing corresponds with the input dataset (from where we want to subset the data). We can use these Boolean values to subset the dataset (only the rows that have a `True` output will be retained).
```

Comment:
put in note admonition? unless it looks silly

Reply:
Looks fine to me!